0006-ffmul-revised.md

```diff
@@ -58,14 +58,8 @@ Note that the actual order of the constraints in the code might not correspond t
 
 Let $n$ be the _native_ field modulus (for example, $n$ is one of the Pasta primes). Our first constraint is to check $(1)$ modulo $n$:
 
-$$
-ab = qf + r\mod n
-$$
-
-or equivalently, which only uses the negated limbs of the foreign modulus:
-
 $$\tag{C1: native}
-ab + q f' - r = 0 \mod n
+ab = qf + r\mod n
```

Comment:
replacing $ab = qf + r$ with $ab + qf' - r = 0$ here was wrong because:
1. it was missing a term $2^{3\ell}q$
2. it used $f'$ which wasn't defined in the text at that point

Reply:
We explain later in the text that the gate implementation uses $2^{3\ell} - f'$ in place of $f$. This is sufficient.

Reply:
I added some other fixes in other parts of the text, I realized they were wrong while doing the implementation.

Reply:
Can you move the definition of `f'`, and inline the `f'` version? The constraints that we enumerate here should be exactly what we're intending to have in the code, there should be no interpretation necessary to check the RFCs constraints against the code.

Reply:
ok that makes sense!